src/components/Trade/hooks/useTradeAmounts.tsx

```diff
@@ -197,6 +193,27 @@ export const useTradeAmounts = () => {
         isSendMax: sendMax ?? isSendMaxFormState,
       })
 
+      const bestSwapperType = tradeQuoteArgs
+        ? (
+            await dispatch(
+              getBestSwapperType.initiate({
+                ...tradeQuoteArgs,
+                feeAsset,
+              }),
+            )
+          ).data
+        : undefined
+
+      const swapperManager = await getSwapperManager(featureFlags)
+      const swappers = swapperManager.swappers
+      const bestTradeSwapper = bestSwapperType ? swappers.get(bestSwapperType) : undefined
+
+      if (!bestTradeSwapper) {
```

Comment:
Can we use `useBestSwapper` here and move this bit inside a `useEffect`, or is the intent to actually await the thunk, meaning we don't want to end up in a first render with the hook data loading and run this?

Reply:
`useBestSwapper` requires a `feeAsset`. 

We don't have a `feeAsset` until we're inside the `useCallback`, and can't call a hook from here. 
Therefore we basically need to re-write the logic in `useBestSwapper` inside the `useCallback`.